lib/web/websocket/connection.js

```diff
@@ -97,11 +99,29 @@ function establishWebSocketConnection (url, protocols, client, handler, options)
         handler.readyState = states.CLOSED
       }
 
-      // 1. If response is a network error or its status is not 101,
+      // 1. If response is a network error or its status is not 101 (or 200 for H2),
```

Comment:
```suggestion
      // 1. If response is a network error or its status is not 101,
```

this comment comes from https://websockets.spec.whatwg.org/#concept-websocket-establish